consensus/bor/bor.go

```diff
@@ -464,18 +464,11 @@ func (c *Bor) verifyCascadingFields(chain consensus.ChainHeaderReader, header *t
 
 	// Verify the validator list match the local contract
 	if IsSprintStart(number+1, c.config.CalculateSprint(number)) {
-		parentBlockNumber := number - 1
 
-		var newValidators []*valset.Validator
-
-		var err error
-
-		for newValidators, err = c.spanner.GetCurrentValidators(context.Background(), chain.GetHeaderByNumber(parentBlockNumber).Hash(), number+1); err != nil && parentBlockNumber > 0; {
-			parentBlockNumber--
```

Comment:
Just out of curiosity @cffls 
I understand the bug about potentially going backwards indefinitely because of `parentBlockNumber--`
But why were we getting `nil` headers with `chain.GetHeaderByNumber` and we have to use `rpc` now?

Reply:
It is still not clear to me why passing explicit block number won't work here. The header was looked up by this function: https://github.com/maticnetwork/bor/blob/3ed22568fd39b4e690ed45a90d9cd44370d01e26/eth/api_backend.go#L65-L76. For some some reason `b.eth.blockchain.GetHeaderByNumber(uint64(number))` will sometimes return nil, but `b.eth.blockchain.CurrentBlock().Header()` won't.

Reply:
Yeah, definitely strange